flaky `test_quiet_close_process[False]`

It's possible this has already been fixed in https://github.com/dask/distributed/pull/6567. If we don't see it again for a couple weeks, let's close this. However, I have a feeling the TimeoutError not being retrieved is some other problem.

```
_______________________ test_quiet_close_process[False] ________________________
processes = False
tmp_path = PosixPath('/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/test_quiet_close_process_False0')
@pytest.mark.parametrize("processes", [True, False])
deftest_quiet_close_process(processes, tmp_path):
withopen(tmp_path / "script.py", mode="w") as f:
            f.write(client_script % processes)
        proc = subprocess.Popen(
            [sys.executable, tmp_path / "script.py"],
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
        )
        out, err = proc.communicate(timeout=10)
assertnot out
>       assertnot err
E       assert not b"Future exception was never retrieved\nfuture: <Future finished exception=TimeoutError('Timeout')>\ntornado.util.TimeoutError: Timeout\n"
distributed/tests/test_client.py:7529: AssertionError
```

https://github.com/dask/distributed/runs/6893247153?check_suite_focus=true#step:11:1378

I'm now getting a different one:


>       assert not out
E       AssertionError: assert not b'/Users/runner/miniconda3/envs/dask-distributed/lib/python3.10/contextlib.py:142: UserWarning: Creating scratch directories is taking a surprisingly long time. This is often due to running workers on a network file system. Consider specifying a local-directory to point workers to write scratch data to a local disk.\n  next(self.gen)\n'

The stdout message is now different from before though:
```

>       assert not out
E       assert not b"Future exception was never retrieved\nfuture: <Future finished exception=TimeoutError('Timeout')>\ntornado.util.TimeoutError: Timeout\n"
```

This test is a canary for all unrelated cleanup and performance problems.
I'm going to build a blacklist of common warnings into it.

--- a/distributed/tests/test_client.py
+++ b/distributed/tests/test_client.py
@@ -7510,16 +7510,29 @@ if __name__ == "__main__":
 
 
 @pytest.mark.slow
+@pytest.mark.flaky(reruns=5, rerun_delay=10, only_rerun="Found blocklist match")
 @pytest.mark.parametrize("processes", [True, False])
 def test_quiet_close_process(processes, tmp_path):
     with open(tmp_path / "script.py", mode="w") as f:
         f.write(client_script % processes)
 
     with popen([sys.executable, tmp_path / "script.py"], capture_output=True) as proc:
-        out, err = proc.communicate(timeout=60)
+        out, _ = proc.communicate(timeout=60)
 
-    assert not out
-    assert not err
+    lines = out.decode("utf-8").split("\n")
+    lines = [stripped for line in lines if (stripped := line.strip())]
+
+    # List of frequent spurious messages that are beyond the scope of this test
+    blocklist = [
+        "Creating scratch directories is taking a surprisingly long time",
+        "Future exception was never retrieved",
+        "tornado.util.TimeoutError",
+    ]
+    lines2 = [line for line in lines if not any(ign in line for ign in blocklist)]
+    # Instant failure for messages not in blocklist
+    assert not lines2
+    # Retry up to 5 times if the only messages are in the blocklist
+    assert not lines, "Found blocklist match, retrying: " + str(lines)
 
 
 @gen_cluster(client=False, nthreads=[])

